dandi/download.py

```diff
@@ -578,27 +583,18 @@ def _download_file(
                 and "sha256" in digests
             ):
                 if key_parts[-1].partition(".")[0] == digests["sha256"]:
+                    lgr.debug(f"{path!r} already exists - matching digest in filename")
                     yield _skip_file("already exists")
                     return
                 else:
                     lgr.debug(
                         "%s is in git-annex, and hash does not match hash on server; redownloading",
                         path,
                     )
-            elif (
-                digests is not None
-                and "dandi-etag" in digests
-                and get_digest(path, "dandi-etag") == digests["dandi-etag"]
-            ):
-                yield _skip_file("already exists")
-                return
-            elif (
-                digests is not None
-                and "dandi-etag" not in digests
-                and "md5" in digests
-                and get_digest(path, "md5") == digests["md5"]
-            ):
-                yield _skip_file("already exists")
+            elif digests is not None and check_digests(path, digests):
+                lgr.debug(f"{path!r} already exists - matching digest")
```

Comment:
I thought, in the light of e.g. https://github.com/dandi/dandi-archive/pull/1750, that we fully avoided using f-strings in logging but at least 2 did sneak in already

```
❯ git grep 'lgr\.[^(]*(f'
dandi/download.py:                lgr.debug(f"{path!r} - same attributes: {same}.  Redownloading")
dandi/upload.py:        lgr.info(f"Found {len(dandi_files)} files to consider")
```

I am not that much of a puritan so wouldn't force a chance here.

Reply:
aha! i didn't realize there was a practical reason for this, thought it was just old style string formatting. did this in https://github.com/dandi/dandi-cli/pull/1364/commits/bc28daab3e1dfc1e5c5912b18137ffa730ccee17